### ISSUE TITLE
hovering above one icon highlights both icons

### ISSUE THREAD

**Issue by 3m4r5** [fawtytoo/activities-icons#4]
it would be nicer if only one icon is highlighted at a time

**Comment by fawtytoo** [comment]
I agree. It helps to know which one you're hovering on before you click it.

Shouldn't be too difficult to implement.

**Comment by fawtytoo** [comment]
After a bit of testing, I find I was a bit hasty in suggesting it would not be difficult. I managed it, but the result is less than pleasing.
At the moment, this uses a single PanelMenu button with 2 icons, which is why they highlight together.
To have them highlight separately would require separating them into 2 PanelMenu buttons. However, with 1 button, they are neatly packed together, but with 2 buttons they are spread apart due to the horizontal spacing on the panel.
I'm willing implement this idea, but I'm going to need time and possibly help with it.

**Comment by 3m4r5** [comment]
that's great, thank you for your efforts and thank you for replying to me I really appreciate it!

**Comment by fawtytoo** [comment]
Closing. There is now a fork of this extension available that separates the icons.

**Comment by fawtytoo** [comment]
I'm reopening this issue due to an idea I found. Credit to Andy Holmes.
[https://stackoverflow.com/questions/45666207/gnome-extension-change-st-icon-image-on-hover](url)
The icons are now animated to increase in size slightly when hovering on them.

Can you please test before I commit this to EGO.

**Comment by 3m4r5** [comment]
@fawtytoo I'm a noob, can you give me the .zip file?

**Comment by fawtytoo** [comment]
Click on `<> Code` at the top of this page, then click the Green Code button, then Download Zip.

**Comment by 3m4r5** [comment]
@fawtytoo turns out I'm not that noob I just didn't know what to test...
anyways, the icons don't get larger for me on Pop!_OS 21.10 (GNOME 40.4.0)
and it's still one button (unlike the fork)

**Comment by 3m4r5** [comment]
also the extension is not compatible with Pop!_OS (the buttons are broken when Pop!_OS extensions are enabled)

**Comment by fawtytoo** [comment]
Did it work before in Pop OS?

And as I said before, turning it into 2 buttons isn't very pretty IMO.

**Comment by 3m4r5** [comment]
> Did it work before in Pop OS?

all versions don't work with Pop!_OS 21.10
but it used to work on previous versions

> And as I said before, turning it into 2 buttons isn't very pretty IMO.

fair enough 👍️

**Comment by fawtytoo** [comment]
I was thinking the problem here is knowing which one the mouse cursor is hovering over if they highlight together. So I thought an animation might be an alternative. I'll keep working on it.

I'll try and test Pop OS at some point. Works fine for me in Debian: GS 3.38 and 41. The trouble with Pop (and Ubuntu), is it's not pure Gnome.

**Comment by fawtytoo** [comment]
I've uploaded another commit for you to try. I think I've cracked it!

There's less difference from the last version that worked for you. So should be ok in Pop OS.

**Comment by 3m4r5** [comment]
> I've uploaded another commit for you to try. I think I've cracked it!

@fawtytoo yeah, the animations are nice!
> There's less difference from the last version that worked for you. So should be ok in Pop OS.

the old version also doesn't work properly on Pop!_OS 21.10 (both versions go back to the main window automatically after less than half a second)
the problem is in the new Pop!_OS 21.10 extensions

**Comment by fawtytoo** [comment]
My first problem in achieving this is, since GS 40, the icons highlight differently. Previous versions just made the icons/text on the panel lighter. Since GS 40, the icons/text are highlighted with a rounded rectangle. Which brings me back to the original problem of increased spacing between icons to allow for those rounded rectangles.

I also notice that since Pop OS 21.04, the Activities button has been replaced with Workspaces/Applications button. I think this is one of their own extensions. It's quite possible that my extension does not properly disable theirs. But that's down to you. I can only deal with a stock Gnome environment.

Personally, I'm ok with having each icon become brighter on hover, with a small increase in size to help visibility. But I don't know what to do with the rounded rectangle around both icons.

Also, can you clarify:


>the old version also doesn't work properly on Pop!_OS 21.10 (both versions go back to the main window automatically after less than half a second)

**Comment by 3m4r5** [comment]
https://user-images.githubusercontent.com/63432932/147973889-e5bf9c66-08df-4106-bb14-f322ebf7c7c5.MOVthis happens if you don't turn off the built in extensions on Pop!_OS 21.10 (I only clicked each button once)
